src/Components/Components/test/Routing/RouteTableFactoryTests.cs

```diff
@@ -284,6 +284,66 @@ public void CanMatchParameterWithConstraint(string template, string contextUrl,
             });
         }
 
+        public static IEnumerable<object[]> CanMatchOptionalParameterWithConstraintCases() => new object[][]
+{
+            new object[] { "/optional/{value:bool?}", "/optional/", null },
+            new object[] { "/optional/{value:datetime?}", "/optional/", null },
+            new object[] { "/optional/{value:decimal?}", "/optional/", null },
+            new object[] { "/optional/{value:decimal?}/another", "/optional//another", null },
```

Comment:
I don't think we'd want this particular pattern to be a match. Seeing `//` in the URL is pretty unexpected.

From experimenting, it seems that server-side ASP.NET Core's router only considers segments to be optional when they are not followed by any non-optional segments.

I can't see this specifically called out in [server-side docs] anywhere, but @pranavkm do you agree this matches the intent?

It's totally expected that the client-side router has fewer matching features than the server-side one, but in cases where we can align the designs that's ideal.

Reply:
Thanks for bringing this up.

With this information, I'm partial to not supporting routes like `weather//blah` as well. For the primary reason that it will greatly simplify the Match logic changes that I made to support optional parameters.

> From experimenting, it seems that server-side ASP.NET Core's router only considers optional segments to match if all subsequent segments are also optional and have no supplied value in the URL. 

From my experience with this implementation, I can understand why they have put in place these constraints.

Reply:
Resolving this as I've added some code that will through an exception if the developer attempts to write a non-optional parameter after an optional one in a route template.